kusto/ingest/internal/properties/properties.go

```diff
@@ -416,6 +416,13 @@ func (i Ingestion) validate() error {
 	return nil
 }
 
+func RemoveQueryParamsFromUrl(url string) string {
```

Comment:
it might be easier to use if the API was `func (s string) RemoveQueryParamsFromUrl() string`
calling would simply be s.RemoveQueryParamsFromUrl()

Reply:
I'm definitely not an expert in Go, but I tried that because it was my preferred approach as well, and it didn't work. IIRC the issue was that I couldn't add a function on the `string` class.

IIRC we do have a 'String` class, and I tried to add this method to there, but I believe it would've required more extensive changes to pass that type of parameter instead of the existing `string`.

Reply:
Yihezkel is right, you can't monkeypatch classes on go (you can't do anything fun on go).
The capital String class is for the string type in kusto (parsing them, escaping them, etc), and should not be used for general purpose.